interfaces/automated-testing/rid/injection.yaml

```diff
@@ -0,0 +1,159 @@
+openapi: 3.0.2
+info:
+  title: Remote ID Test Data Injection
+  version: 0.0.1
+  description: >-
+    This interface is implemented by every Service Provider wishing to be tested by the automated testing framework.
+    The automated testing suite calls this interface to inject flight-related test data into the Service Provider system under test.
+paths:
+  /tests/{test_id}:
+    parameters:
+      -
+        examples:
+          default:
+            value: '2979bd18-7f06-441c-bda6-e82c841c35d6'
+        name: test_id
+        description: >-
+          The ID of the test.
+        schema:
+          type: string
+        in: query
+        required: true
+    put:
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateTestParameters'
+        required: true
+      responses:
+        '200':
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ChangeTestResponse'
+          description: Requested test was created successfully.
+        '409':
+          description: Test already exists.
+      operationId: createTest
+      summary: Create test
+      description: >-
+        Request the creation of one or more logical flights based on injection of the provided test data.
+        The host Service Provider's test data injector may modify fields, such as `details.id`, as necessary to work with their system.
+        The host Service Provider's test data injector should modify as few fields as practical, and change modified fields as little as practical.
+        The `injection_id` fields may not be modified.
+  /tests/{test_id}/{version}:
+    parameters:
+      -
+        examples:
+          default:
+            value: '2979bd18-7f06-441c-bda6-e82c841c35d6'
+        name: test_id
+        description: >-
+          The ID of the test.
+        schema:
+          type: string
+        in: query
+        required: true
+      -
+        name: version
+        description: >-
+          Current version of the test.
+        schema:
+          type: string
+        in: query
+        required: true
+    delete:
+      responses:
+        '200':
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DeleteTestResponse'
+          description: Requested test was deleted successfully.
+      operationId: deleteTest
+      summary: Delete test
+      description: Remove all injected test data associated with this test from the host Service Provider's system under test.
+components:
+  schemas:
+    CreateTestParameters:
+      description: 'A complete set of data to be injected into a Service Provider system under test.'
+      required:
+        - requested_flights
+      type: object
+      properties:
+        requested_flights:
```

Comment:
Are all flight positions of all flights submitted to the endpoint in one `Create Test` request? A concern is that request body data might be too large and we might to set some limit as to how many flights can be created. e.g. one SP can create max two flights (or maybe just one).

Reply:
Yes, the intention here was to include all injected data for all flights for a single Service Provider for a single test in a single call.  The data is carried in a request body rather than URL, so we're not subject to the ~2k limit for URLs.  Is there a reasonable server implementation that would have trouble with request bodies of this size?  If we have a 10 minute test (seems much longer than necessary) with 3 flights per SP (seems high; I think only 1 is needed for currently-identified tests), a standard 1 Hz update rate would result in a request body no larger than about 1 MB if I calculate correctly.  Do you expect a 1 MB (or even 2 MB) request body to be a problem for any web server an SP is likely to use?

Also, just to clarify, the automated testing system would be requesting the creation of these flights (not the SP).  So, with the current set of tests described in the SUSI document, I believe we can limit the number of flights requested of each SP to 1 by simply choosing to only request one flight per SP per test.

Reply:
No problems, having one flight per SP is sounds OK since we may not "simultaneously" test it like the SUSI checkouts. In this case, I dont think it should be a problem.

Reply:
Since the automated testing system is the one requesting the creation of flights, it would also be easy to request more than one flight per USS, though I don't think any of the current tests require this as described.  The most important aspect to test "simultaneously" is for each SP to have at least one flight in an area at the same time.  Multiple simultaneous flights for a single SP would probably also be nice to test, but I think the simultaneity with multiple SPs is the most important dimension to exercise.

Reply:
Agreed, we have that piece covered, with the multiple tracks that can be assigned to a different SP.